### ISSUE TITLE
Memory Leak with importance sampling.

### ISSUE THREAD

**Issue by kris-singh** [keras-team/keras#9583]
While implementing the following loss based importance sampling scheme for PTB language modelling task. There is memory leak error that happens. PFA the link to the [code](https://gist.github.com/kris-singh/5791a32e9b8132bae3d76794a19a649c. I do not understand how this is happening as I use batch_size=20 and sequence_length=20 and i have around 12gb GPU memory.
)- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps

- [x] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by Dref360** [comment]
It's a gpu memory issue or a RAM issue?

**Comment by kris-singh** [comment]
I am not sure but i think it's the RAM memory issues.
dmesg gives this as output
Out of memory: Kill process 21436 (python) score 985 or sacrifice child
Killed process 21436 (python) total-vm:82999692kB, anon-rss:31931456kB, file-rss:83340kB

**Comment by Dref360** [comment]
Each time you sample, you create an op in the graph. This is the cause of your memory issue.
Define placeholders and ops and then call sess.run to get your value.
https://gist.github.com/kris-singh/5791a32e9b8132bae3d76794a19a649c#file-temp-py-L88

**Comment by kris-singh** [comment]
I am not entirely sure how to do that. Do you mean something like this
```
class SelectLoss:
    """
        Selection based on Loss values of samples.
        No need of rejection sampling.
    """

    def __init__(self, loss):
        """
        :param loss: loss function
        :param x_train: training data
        :param y_train: training labels
        """
        self.loss = loss
        self.x_train = x_train
        self.y_train = y_train
        self.idx = np.random.choice (np.arange (0, self.x_train.shape[0]), size=args.fwd_batch_size, replace=False)
        self.res = model.predict_proba (self.x_train[self.idx])
        self.temp = K.get_value(sparse_categorical_crossentropy(tf.convert_to_tensor(self.y_train[idx], np.float32), tf.convert_to_tensor(self.res)))
        self.temp = self.temp / np.sum (self.temp)
        self._sample = K.function([self.res, self.y_train], self.temp)


    def sample(self, x_train, y_train):
        """
            Sort the loss values of the training samples.
            Variables
        """
        return self._sample(x_train, y_train)
```
Or if not this could you give a example how you would do what you are saying.
